Implement KNNRegressor

bojanbla opened the pull request:
Summary:

- Implement multi variable linear regression using **KNN classifier**
- Short description added in **README.md**

bojanbla committed:
Implement KNNRegressor for multi variable linear regression

drmarjanovic commented on ex-02-multi-var-lin-reg/knn_regressor.py:
```diff
@@ -0,0 +1,121 @@
+        for i in range(len(x)):
+            distance += (x[i] - y[i]) ** 2
+
+        return sqrt(distance)
```
You can use **map** with **zip** here.

stkeky replied:
Change was made to use `map` and `reduce`.

drmarjanovic commented on ex-02-multi-var-lin-reg/knn_regressor.py:
```diff
@@ -0,0 +1,121 @@
+        """
+        distance = 0
+        for i in range(len(x)):
+            distance += abs(x[i] - y[i])
```
Also, you can use **map** and **zip** here. 😄

stkeky replied:
Change was made to use `map` and `reduce`.

drmarjanovic commented on ex-02-multi-var-lin-reg/test.py:
```diff
@@ -0,0 +1,57 @@
+    hello()
+
+    # Read training data from file
+    df = pd.read_csv('data/train.csv')
```
Use more clear name for variables. What does this mean? 😄

stkeky replied:
`df` is 'standard' short-name for `pandas.DataFrame` so it's not that bad 😃

drmarjanovic commented on ex-02-multi-var-lin-reg/test.py:
```diff
@@ -0,0 +1,57 @@
+
+    x_train, y_train = train.ix[:, :-1], train.ix[:, -1]
+    x_validation, y_validation = train.ix[:, :-1], train.ix[:, -1]
+
```
I'm not sure that we now need to split data. Validation set will be provided from his side. Maybe, you can make additional parameter in **args** for this. Maybe we should use all this provided data for training.

stkeky replied:
I agree, TA provided separate `*.csv` files.

stkeky committed:
Makeup changes

Changed distance calucaltions to use map and reduce methods. Lowered all
column names for easier manipulation. Extracted binary and nominal
fields to variables to prevent code repetition and possible mistakes.

bojanbla committed:
Change validation set

drmarjanovic committed:
Code cleanup

drmarjanovic committed:
Change to median instead of mean in data normalization

drmarjanovic committed:
Add support for choosing percent of validation set

drmarjanovic committed:
Update README.md
